### PR TITLE
Add float8 types

### PR DESCRIPTION
**Pull request by niboshi**
Adds float8 support

### CODE REVIEW

File: pytorch_pfn_extras/onnx/pfto_exporter/export.py [hunk 1]
```diff
@@ -168,6 +168,14 @@ def _type_to_proto(t: torch._C.TensorType) -> onnx.TypeProto:
 
     if t.scalarType() is None:
         ret.tensor_type.elem_type = onnx.TensorProto.DataType.UNDEFINED  # type: ignore[attr-defined]
+    elif t.scalarType() == "Float8_e4m3fn":
+        ret.tensor_type.elem_type = int(  # type: ignore
+            sym_hel._C_onnx.TensorProtoDataType.FLOAT8E4M3FN
```
Comment (take-cheeze) [inline pytorch_pfn_extras/onnx/pfto_exporter/export.py:173]: This should referred from onnx package instead for old torch
```suggestion
            onnx.TensorProto.DataType.FLOAT8E4M3FN  # type: ignore[attr-defined]
```

Comment (niboshi) [reply pytorch_pfn_extras/onnx/pfto_exporter/export.py]: Thank you. Fixed as such: 239ae63d85d3bbd76fe380cddaf07cf6f14d6b48

File: pytorch_pfn_extras/onnx/pfto_exporter/export.py [hunk 2]
```diff
@@ -221,6 +229,8 @@ def onnx_node_doc_string(onnx_node: torch._C.Node, torch_node: torch._C.Node) ->
     torch.float16: onnx.TensorProto.DataType.FLOAT16,  # type: ignore[attr-defined]
     torch.complex64: onnx.TensorProto.DataType.COMPLEX64,  # type: ignore[attr-defined]
     torch.complex128: onnx.TensorProto.DataType.COMPLEX128,  # type: ignore[attr-defined]
+    torch.torch.float8_e4m3fn:  onnx.TensorProto.DataType.FLOAT8E4M3FN,  # type: ignore[attr-defined]
+    torch.torch.float8_e5m2:  onnx.TensorProto.DataType.FLOAT8E5M2,  # type: ignore[attr-defined]
```
Comment (take-cheeze) [inline pytorch_pfn_extras/onnx/pfto_exporter/export.py:233]: https://github.com/pfnet/pytorch-pfn-extras/actions/runs/6702797789/job/18212320599?pr=781#step:8:379
```suggestion
    torch.float8_e4m3fn:  onnx.TensorProto.DataType.FLOAT8E4M3FN,  # type: ignore[attr-defined]
    torch.float8_e5m2:  onnx.TensorProto.DataType.FLOAT8E5M2,  # type: ignore[attr-defined]
```

Comment (niboshi) [reply pytorch_pfn_extras/onnx/pfto_exporter/export.py]: Thanks. It seems we need to check versions of pytorch and onnx: 98c54754b0e1a5b78c2d237e91874f09f47056c2

**Review by take-cheeze — APPROVED** [review]
LGTM! 👍

**Pull request merged**